stdlib/public/core/ArrayBuffer.swift

```diff
@@ -567,6 +571,78 @@ extension _ArrayBuffer {
     }
   }
 
+  @inlinable @_alwaysEmitIntoClient
+  static var associationKey: UnsafeRawPointer {
+    //We never dereference this, we just need an address to use as a unique key
+    UnsafeRawPointer(Builtin.addressof(&_swiftEmptyArrayStorage))
+  }
+  
+  @inlinable @_alwaysEmitIntoClient
+  internal func getAssociatedBuffer() -> _ContiguousArrayBuffer<Element>? {
+    let getter = objc_getAssociatedObject as @convention(c)(
+      Any,
+      UnsafeRawPointer
+    ) -> Any?
+    let typedGetter = unsafeBitCast(
+      getter,
+      to: (@convention(c)(
+        AnyObject,
+        UnsafeRawPointer
+      ) -> AnyObject?).self
+    )
+    if let assoc = typedGetter(
+      _storage.objCInstance,
+      _ArrayBuffer.associationKey
+    ) {
+      let buffer = unsafeBitCast(
+        assoc,
+        to: _ContiguousArrayStorage<Element>.self
+      )
+      return _ContiguousArrayBuffer(buffer)
+    }
+    return nil
+  }
+  
+  @inlinable @_alwaysEmitIntoClient
+  internal func setAssociatedBuffer(_ buffer: _ContiguousArrayBuffer<Element>) {
+    let setter = unsafeBitCast(getSetAssociatedObjectPtr(), to: (@convention(c)(
+      AnyObject,
+      UnsafeRawPointer,
+      AnyObject?,
+      UInt
+    ) -> Void).self)
+    setter(
+      _storage.objCInstance,
+      _ArrayBuffer.associationKey,
+      buffer._storage,
+      0o1401 //OBJC_ASSOCIATION_RETAIN
```

Comment:
`OBJC_ASSOCIATION_RETAIN_NONATOMIC` will save you a retain/autorelease when getting the value. The atomic version just protects you from a concurrent set replacing the object while you're retrieving it, and that won't happen here. I expect the performance difference would be pretty small, though, since the autorelease will be elided.